src/sap/fhir/model/r4/FHIRModel.js

```diff
@@ -1211,7 +1220,7 @@ sap.ui.define([
 	 * @since 1.0.0
 	 */
 	FHIRModel.prototype.hasResourceTypePendingChanges = function(sResourceType) {
-		return this.mChangedResources[sResourceType] !== undefined;
+		return this.mChangedResources[sResourceType] !== undefined && Object.keys(this.mChangedResources[sResourceType]).length > 0;
```

Comment:
why the second condition is needede?

Reply:
we are just doing a delete of particular id from mChangedResources object.. which might lead empty object Patient {}

Reply:
Thx